tfx/examples/bert/cola/bert_cola_utils.py

```diff
@@ -0,0 +1,182 @@
+# Lint as: python2, python3
+# Copyright 2019 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Python source file include cola pipeline functions and necessary utils."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import sys
+
+from typing import List, Text
+import tensorflow as tf
+import tensorflow_transform as tft
+import tensorflow_hub as hub
+import tensorflow_addons as tfa
+from tfx.components.trainer.executor import TrainerFnArgs
+
+_bert_utils_root = os.path.join(os.environ['HOME'], 'bert', 'utils')
+sys.path.append(_bert_utils_root)
+
+# pylint: disable=wrong-import-position
+from bert_tokenizer_utils import SpecialBertTokenizer
+from bert_models import BertForClassification
+# pylint: enable=wrong-import-position
+
+_TRAIN_BATCH_SIZE = 32
+_EVAL_BATCH_SIZE = 32
+_LABEL_KEY = "label"
+_BERT_LINK = "https://tfhub.dev/tensorflow/bert_en_cased_L-12_H-768_A-12/2"
+_MAX_LEN = 256
+_EPOCHS = 1
+
+
+def _gzip_reader_fn(filenames):
+  """Small utility returning a record reader that can read gzip'ed files."""
+  return tf.data.TFRecordDataset(
+      filenames,
+      compression_type='GZIP')
+
+
+def _tokenize(feature):
+  """Tokenize the two sentences and insert appropriate tokens"""
+  tokenizer = SpecialBertTokenizer(_BERT_LINK)
+  return tokenizer.tokenize_single_sentence(
+      tf.reshape(feature, [-1]),
+      max_len=_MAX_LEN)
+
+
+def preprocessing_fn(inputs):
+  """tf.transform's callback function for preprocessing inputs.
+
+  Args:
+    inputs: map from feature keys to raw not-yet-transformed features.
+
+  Returns:
+    Map from string feature key to transformed feature operations.
```

Comment:
Should be "Map from string feature key to transformed feature Tensors/SparseTensors."
Similarly below for the other preprocessing_fn.

Reply:
I think the output is strictly Tensors. But fixed!